File: src/Website/Models/SubmitEpisode/SubmittedPlayedCharacter.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using Website.Models.Database.Enums;

namespace Website.Models.SubmitEpisode
{
    public class SubmittedPlayedCharacter
    {
        [Required]
        public string CharacterId { get; set; } = string.Empty;

        [Required]
        public GameMode GameMode { get; set; }

        [Required]
        public List<SubmittedPlayedFloor> PlayedFloors { get; set; } = new List<SubmittedPlayedFloor>();
    }
}

File: src/Website/Ts/Models/submitted-played-character.ts
import { SubmittedPlayedFloor } from "./submitted-played-floor";
import { GameMode } from "../Enums/game-mode";

export interface SubmittedPlayedCharacter {
    CharacterId: string,
    GameMode: GameMode,
    PlayedFloors: Array<SubmittedPlayedFloor>,
    Seed: string | null |undefined
}